### PR TITLE
fix(vault): show icon in controller listing if all controllers are configured but secrets are not migrated

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Icons now show next to controllers if they are configured to use Vault but secrets have not been migrated

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)

### QA steps

- Either on your system or in a Multipass instance, ensure you've installed the latest MAAS snap with a test db:
   - `sudo snap install maas --channel=latest/edge`
   - `sudo snap install maas-test-db --channel=latest/edge`
- Initialise maas and create an admin user
- Install Vault: `sudo snap install vault` and jq: `sudo apt install jq`
- Run Vault: `vault server -dev -dev-root-token-id=root-token`
- In a separate terminal, run this script to set up vault: https://git.launchpad.net/maas/plain/utilities/configure-vault (you'll need to download it first). You can run it like this: `./configure-vault`, or like this: `sudo -E bash configure-vault setup`, `sudo -E bash configure-vault create-approle role-name`
   - This may require you to set some environment variables. The values you need will be in your terminal output where Vault is running, e.g. `export VAULT_TOKEN='root-token'`
   - `export VAR_NAME="varvalue"`
- Once the script has run successfully, run the command that it gives you. This will configure the primary controller to use Vault for secret storage.
- Ensure your MAAS-UI instance (this branch) is pointing to your local MAAS instance (you can set the MAAS_URL in a .env.local file)
- Go to the controller listing
- Ensure that the controller has a shield icon next to its type, with a tooltip.

### CODE REVIEW

**Comment by ndv99** [comment]
@petermakowski I've put the QA steps in here detailing how to set up Vault, but we'll go through it in our meeting later today

File: src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx [hunk 1]
```diff
@@ -99,7 +97,7 @@ const generateRows = ({
             <span className="u-truncate">
               {controller.node_type === NodeType.REGION_CONTROLLER ||
               controller.node_type === NodeType.REGION_AND_RACK_CONTROLLER ? (
-                vaultEnabled ? (
+                vaultEnabled === true ? (
```
Comment (petermakowski) [inline src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx:101]: What's the reason for this change? `vaultEnabled` is of boolean type anyway, TS will ensure this can be either true or false.

File: src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx [hunk 2]
```diff
@@ -120,27 +118,25 @@ const generateRows = ({
...
+                      <p id="tooltip-description">
+                        Vault is configured on this controller. <br />
+                        Once all controllers are configured, migrate the
+                        secrets. <br />
+                        <a href={docsUrls.vaultIntegration}>
+                          Read more about Vault integration
+                        </a>
+                      </p>
+                    }
+                  />
```
Comment (petermakowski) [inline src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx]:
```suggestion
                  <Tooltip
                    message={
                      <p id="tooltip-description">
                        Vault is configured on this controller. <br />
                        Once all controllers are configured, migrate the
                        secrets. <br />
                        <a href={docsUrls.vaultIntegration}>
                          Read more about Vault integration
                        </a>
                      </p>
                    }
                  >
                    <Button appearance="base" hasIcon small type="button">
                      <Icon
                        aria-describedby="tooltip-description"
                        data-testid="vault-icon"
                        name="security"
                      />
                    </Button>
                  </Tooltip>
```

File: src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx (hunk 1 above)
Comment (ndv99) [reply src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx:101]: Ah that was left in there from some testing and bug fixing. Will change

File: src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx [hunk 3]
```diff
@@ -120,39 +117,33 @@ const generateRows = ({
...
+                      iconProps={{
+                        "data-testid": "vault-icon",
+                        "aria-describedby": "tooltip-description-warning",
+                      }}
                       message={
-                        <p id="tooltip-description">
+                        <p id="tooltip-description-warning">
                           Missing Vault configuration.
                           <br />
                           <a href={docsUrls.vaultIntegration}>
```
Comment (petermakowski) [inline src/app/controllers/views/ControllerList/ControllerListTable/ControllerListTable.tsx:149]: Sorry, I missed this in the first review.

We have a special class to be used for links in tooltips (which have a dark background) to get accessibility compliant colour contrast. Also, external docs links are usually opened in a new tab. `className="is-on-dark"`.

Could you please update along with other links in this file.
 
```suggestion
                          <Link
                            href={docsUrls.vaultIntegration}
                            rel="noreferrer noopener"
                            target="_blank"
                          >
```

**Review by petermakowski — APPROVED** [review]
Awesome. Much better!
